pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -426,6 +426,8 @@ boolean markAckForBatchMessage(BatchMessageIdImpl batchMessageId, AckType ackTyp
                 && !batchMessageId.getAcker().isPrevBatchCumulativelyAcked()) {
```

Comment:
Why should we add condition of !batchMessageId.getAcker().isPrevBatchCumulativelyAcked()

Reply:
Seems here, is to ack the previous whole batch message. If it is already acked, we should not send ack again.